File: Algorithm/Programmers/Python/Level 2/2 x n 타일링.py
def solution(n):
    if n == 1 :
        return 1
    if n == 2 :
        return 2
    
    dpTable = [0, 1, 2]
    for i in range(3, n + 1) :
        dpTable.append((dpTable[i - 1] + dpTable[i - 2]) % 1000000007)
    return dpTable[n]

File: Algorithm/Programmers/Java/Level 2/2_x_n_타일링.java
class Solution {
    public int solution(int n) {
        if(n == 1) {
            return 1;
        }
        
        if(n == 2) {
            return 2;
        }
        
        int a = 1;
        int b = 2;
        for(int i = 3; i <= n; i++) {
            int tmp = (a + b) % 1000000007;
            a = b;
            b = tmp;
        }
        return b;
    }
}